File: DSA/DSSession3/PriorityQueue/Queue.java
package GET2018.DSA.DSSession3.PriorityQueue;

/**
 * This interface provides various operations on Queue
 * 
 * @author Mohit Sharma
 *
 */
public interface Queue {

	/**
	 * Adds element to Queue
	 * 
	 * @param newElement
	 * @param priority
	 * @return
	 */
	public boolean enqueue(String newElement, int priority);

	/**
	 * Removes Front Element from Queue
	 * 
	 * @return
	 */
	public String dequeue();

	/**
	 * Peeks in Queue
	 * 
	 * @return
	 */
	public String peek();

	/**
	 * Checks whether Queue is Full or not
	 * 
	 * @return
	 */
	public boolean isFull();

	/**
	 * Checks whether Queue is Empty or not
	 * 
	 * @return
	 */
	public boolean isEmpty();
}

File: UIDD/UIDD-Session8/queue.js
//Creates Queue
class Queue {
	constructor() {
		this.elements = [];
	}
}

//Checks whether Queue is Empty or not
Queue.prototype.isEmpty = function () {
	if (this.elements.length == 0) {
		return true;
	}
	return false;
}

//Adds Element to Queue
Queue.prototype.enqueue = function (data) {
	this.elements.push(data);
}
//Deletes Element from Queue
Queue.prototype.dequeue = function () {
	if (this.isEmpty()) {
		return "Queue Underflow";
	}
	return this.elements.shift();
}

//Traverses Queue
Queue.prototype.display = function () {
	for (var i = 0; i < this.elements.length; i++) {
		console.log(this.elements[i]);
	}
}